docs/docs/handling-events.md

```diff
@@ -139,3 +139,56 @@ class LoggingButton extends React.Component {
 ```
 
 The problem with this syntax is that a different callback is created each time the `LoggingButton` renders. In most cases, this is fine. However, if this callback is passed as a prop to lower components, those components might do an extra re-rendering. We generally recommend binding in the constructor or using the property initializer syntax, to avoid this sort of performance problem.
+
+You can also use the `bind` method to pass arguments to your event handlers, which is particularly useful for knowing which item in a list was clicked. As shown above, the first argument for the bind method will be the scope you want bound to your function.  The last argument received by your event handler will always be the synthetic event.  You have the ability to pass any number of arguments after `this` and before the event to your event handler:
+
+
+```js{2-5,14}
+class List extends React.Component {
+  handleListClick (item, i, event) {
+    console.log(`this item was clicked: ${item}`);
+    console.log(`it is item number ${i}`);
+  }
+
+  render () {
+    const listItems = ['Item One', 'Item Two', 'Item Three'];
+    return (
+      <ul>
+        {listItems.map((item, i) => {
+          return (
+            <li key={i}>
+              <a onClick={this.handleListClick.bind(this, item, i)}>{item}</a>
+            </li>
+          )
+        })}
+      </ul>
+    )
+  }
+}
+```
+
+There is one more way to do event binding that does not cause a `.bind()` to be called every on every render.  You can set your event handler to a method that is an arrow function which returns nested arrow function.  The ES6 syntax looks a little funny at first but it totally makes sense once you get what it is doing.  The first arrow function is given the arguments you would like to access in your handler. It gets invoked immediately during the first render and returns the inner arrow function.  The inner arrow function is what will be handling the click.  The inner arrow function has access to the parameters passed to the outer arrow function. In JavaScript, nested functions have access to variables declared in their outer scope.  It will only be called when the event occurs giving you a clean event handler:
```

Comment:
After reading this over a few times, I think this is a little too complicated for this section. It's not a common pattern in my experience and also is something that someone familiar with JS functions should be able to figure out themselves if they want. Can we remove it?